File: cliente.py
import socket

direccion_ip_servidor = input("IP del servidor (Enter para localhost): ")
if direccion_ip_servidor == "":
    direccion_ip_servidor = "127.0.0.1"

cliente_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
cliente_socket.connect((direccion_ip_servidor, 8080))

peticion_inicial = "index.txt"
cliente_socket.send(peticion_inicial.encode('utf-8'))
respuesta_servidor = cliente_socket.recv(4096).decode('utf-8')
print("\n" + respuesta_servidor + "\n")

while True:
    comando_usuario = input("-> ")
    
    if comando_usuario.strip() == "":
        continue
        
    cliente_socket.send(comando_usuario.encode('utf-8'))
    
    if comando_usuario == ">END":
        break
        
    respuesta_servidor = cliente_socket.recv(4096).decode('utf-8')
    print("\n" + respuesta_servidor + "\n")

cliente_socket.close()

File: cliente.cpp
#include <iostream>
#include <winsock2.h> // Librería principal para Sockets en Windows

#pragma comment(lib, "ws2_32.lib") // Instrucción para enlazar la librería de red

// Comando de compilación en windows (CMD):  g++ cliente.cpp -o cliente.exe -lws2_32

int main() {
    // 1. Inicializar Winsock (Requisito de Windows)
    WSADATA wsaData;
    WSAStartup(MAKEWORD(2, 2), &wsaData);

    // 2. Crear socket
    SOCKET sock = socket(AF_INET, SOCK_STREAM, 0);

    // 3. Configurar dirección
    sockaddr_in serv_addr;
    serv_addr.sin_family = AF_INET;
    serv_addr.sin_port = htons(8080);
    serv_addr.sin_addr.s_addr = inet_addr("127.0.0.1");

    // 4. Conectar
    connect(sock, (struct sockaddr*)&serv_addr, sizeof(serv_addr));
    std::cout<<"conexion establecida con el servidor ..."<<std::endl;

    // 5. Enviar y recibir
    char buffer[1024] = {0};
    std::string nombre_archivo;
    std::cout << "Nombre del archivo: ";
    std::cin >> nombre_archivo;
    
    send(sock, nombre_archivo.c_str(), nombre_archivo.length(), 0);
    recv(sock, buffer, 1024, 0);

    std::cout << "Respuesta:\n" << buffer << std::endl;

    // 6. Limpieza
    closesocket(sock);
    WSACleanup();
    return 0;
}